qiskit_experiments/data_processing/data_processor.py

```diff
@@ -54,23 +54,29 @@ def append(self, node: DataAction):
         """
         self._nodes.append(node)
 
-    def __call__(self, datum: Dict[str, Any]) -> Any:
+    @property
+    def is_trained(self) -> bool:
+        """Return True if all nodes of the data processor have been trained."""
+        return all(node.is_trained for node in self._nodes)
+
+    def __call__(self, datum: Dict[str, Any], **options) -> Tuple[Any, Any]:
```

Comment:
Since this options will be passed to every node, is it expected that nodes filter only the ones they need? What if certain nodes take the same option name kwarg? (this is similar to current problem with kwargs in experiment and analysis classes)

Reply:
Sounds like this needs some more discussion. In the interest of merging the SVD I removed `**options` here: e7884bc